browser/src/layer/tile/CommentListSection.ts

```diff
@@ -1681,6 +1698,63 @@ export class CommentSection extends CanvasSectionObject {
 		return newArray;
 	}
 
+	private addUpdateChildGroups() {
+		var parentCommentList: Array<any> = [];
+		for (var i = 0; i < this.sectionProperties.commentList.length; i++) {
+			var comment = this.sectionProperties.commentList[i];
+			comment.sectionProperties.children = [];
+			if (comment.sectionProperties.data.parent !== '0')
+			{
+				if (!parentCommentList.includes(comment.sectionProperties.data.parent))
+					parentCommentList.push(comment.sectionProperties.data.parent);
+			}
+		}
+
+		for (var i = 0; i < parentCommentList.length; i++) {
+			var parentComment;
+			for (var j = 0; j < this.sectionProperties.commentList.length; j++) {
+				if (this.sectionProperties.commentList[j].sectionProperties.data.id === parentCommentList[i]) {
+					parentComment = this.sectionProperties.commentList[j];
+					break;
+				}
+			}
+
+			if (parentComment) {
+				for (var j = 0; j < this.sectionProperties.commentList.length; j++) {
+					if (this.sectionProperties.commentList[j].sectionProperties.data.parent === parentCommentList[i])
+						parentComment.sectionProperties.children.push(this.sectionProperties.commentList[j]);
+				}
+			}
+			else
+				console.warn('Couldn\'t find parent comment.');
+		}
+	}
+
+	private addChildrenCommentsToList(comment: any, newOrder: Array<any>) {
+		comment.sectionProperties.children.forEach(function(element: any) {
+			newOrder.push(element);
+			if (element.sectionProperties.children.length > 0)
+				this.addChildrenCommentsToList(element, newOrder);
+		}.bind(this));
+	}
+
+	private orderTextComments() {
+		var newOrder = [];
+
+		for (var i = 0; i < this.sectionProperties.commentList.length; i++) {
+			var comment = this.sectionProperties.commentList[i];
+
+			if (comment.sectionProperties.data.parent === '0') {
```

Comment:
this is repeated many times (check if parent is '0'), this has some special meaning I guess.
Would be good to create separate function for that which will name that check - will be easier to understand the code.

like `isRootComment(comment)` - or what does that mean...